src/github.com/matrix-org/dendrite/clientapi/auth/auth.go

```diff
@@ -62,7 +62,7 @@ func VerifyAccessToken(req *http.Request, deviceDB DeviceDatabase) (device *auth
 	if err != nil {
 		if err == sql.ErrNoRows {
 			resErr = &util.JSONResponse{
-				Code: 403,
+				Code: 401,
```

Comment:
This matches synapse behaviour, and is what sytest expects

Reply:
Indeed: https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#client-authentication

"When credentials are required but missing or invalid, the HTTP call will return with a status of 401 and the error code, M_MISSING_TOKEN or M_UNKNOWN_TOKEN respectively."

so, er, shouldn't we be returning `M_UNKNOWN_TOKEN` ?